packages/sampler/stories/moonstone-stories/VirtualGridList.js

```diff
@@ -51,16 +52,28 @@ const
 		);
 	};
 
-for (let i = 0; i < 1000; i++) {
+const updateDataSize = (dataSize) => {
 	const
-		count = ('00' + i).slice(-3),
-		text = `Item ${count}`,
-		subText = `SubItem ${count}`,
-		color = Math.floor((Math.random() * (0x1000000 - 0x101010)) + 0x101010).toString(16),
-		source = `http://placehold.it/300x300/${color}/ffffff&text=Image ${i}`;
+		itemNumberDigits = dataSize > 0 ? ((dataSize - 1) + '').length : 0,
+		headingZeros = Array(itemNumberDigits).join('0');
 
-	items.push({text, subText, source});
-}
+	items.length = 0;
+
+	for (let i = 0; i < dataSize; i++) {
+		const
+			count = (headingZeros + i).slice(-itemNumberDigits),
+			text = `Item ${count}`,
+			subText = `SubItem ${count}`,
+			color = Math.floor((Math.random() * (0x1000000 - 0x101010)) + 0x101010).toString(16),
+			source = `http://placehold.it/300x300/${color}/ffffff&text=Image ${i}`;
+
+		items.push({text, subText, source});
+	}
+
+	return dataSize;
```

Comment:
I could see the following warnnings.
```
The specified value "NaN" is not a valid number. The value must match to the following regular expression: -?(\d+|\d+\.\d+|\.\d+)([eE][-+]?\d+)?
```
Do we need to return 0 if `dataSize` equals null?

Reply:
I've tested all changed samples with various values including 'no value', but I could not see that error. Could you let me know how can I reproduce it?

Reply:
It turns out that it is not from our code but from a knob-addon.